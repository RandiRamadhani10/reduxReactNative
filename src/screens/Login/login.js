import React, {useState, useEffect} from 'react';
import {
  View,
  Text,
  TextInput,
  StyleSheet,
  Image,
  Dimensions,
  KeyboardAvoidingView,
  TouchableOpacity,
  Alert,
} from 'react-native';
import {useSelector, useDispatch} from 'react-redux';
import getLogin from '../../data/redux/screens/login/action';
const screen = Dimensions.get('screen');
const Login = ({navigation}) => {
  const dispatch = useDispatch();
  const logins = useSelector(state => state.login);
  const [email, setEmail] = useState();
  const [password, setPassword] = useState();
  const runLogin = (dataEmail, dataPassword) => {
    const data = {email: dataEmail, password: dataPassword};
    dispatch(getLogin(data));
    logins.isLogin && navigation.replace('Home');
    logins.isMsg && Alert.alert('Email dan Password salah');
  };
  // const validateLogin = () => {
  //   logins.isLogin && navigation.replace('Home');
  //   logins.isMsg && Alert.alert('Email dan Password salah');
  // };
  // useEffect(() => {
  //   validateLogin();
  // }, [logins]);
  return (
    <View style={styles.parent}>
      <Image
        style={styles.img}
        source={require('../../assets/logoGedebook.png')}
      />
      <Text style={styles.text}>LOGIN</Text>
      <KeyboardAvoidingView
        style={{width: screen.width * 0.8, alignItems: 'center'}}>
        <TextInput
          testID="email-input"
          style={styles.input}
          onChangeText={setEmail}
          value={email}
          placeholder="Email"
          keyboardType="default"
        />
        <TextInput
          testID="password-input"
          style={styles.input}
          onChangeText={setPassword}
          value={password}
          secureTextEntry={true}
          placeholder="Password"
          keyboardType="default"
        />
        <TouchableOpacity
          testID="button-login"
          style={styles.btn}
          onPress={() => {
            runLogin(email, password);
          }}>
          <Text style={{color: 'white', fontSize: 20}}>Login</Text>
        </TouchableOpacity>
      </KeyboardAvoidingView>
      <Text style={{marginTop: 30}}>Dont have account ?</Text>
      <TouchableOpacity
        onPress={() => {
          navigation.navigate('Register');
        }}>
        <Text style={{color: 'black', fontSize: 20, fontWeight: 'bold'}}>
          Register
        </Text>
      </TouchableOpacity>
    </View>
  );
};
const styles = StyleSheet.create({
  parent: {flex: 1, justifyContent: 'center', alignItems: 'center'},
  img: {width: screen.width * 0.7, height: screen.width * 0.7},
  text: {fontSize: 20, color: 'black', fontWeight: 'bold'},
  input: {
    width: '100%',
    height: 40,
    margin: 12,
    borderWidth: 1,
    padding: 10,
  },
  btn: {
    padding: 10,
    margin: 5,
    borderRadius: 30,
    backgroundColor: 'black',
    alignItems: 'center',
    width: screen.width * 0.3,
  },
});
export default Login;
