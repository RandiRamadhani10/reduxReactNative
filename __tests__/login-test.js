import 'react-native';
import React from 'react';
import {connectAdvanced, useSelector} from 'react-redux';
import Login from '../src/screens/Login/login';
import {render, fireEvent, act} from '@testing-library/react-native';
import {miniSerializeError} from '@reduxjs/toolkit';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(() => jest.fn()),
  useSelector: jest.fn(state => {
    state({});
    return {email: '', password: '', token: null, isLogin: false, isMsg: false};
  }),
}));

describe('render login screen', () => {
  it('renders correctly', () => {
    const components = render(<Login />).toJSON();

    expect(components).toMatchSnapshot();
  });
});

describe('getButton', () => {
  // const data = {email: 'aaa@mail.com', password: 'sangat rahasia'};
  it('button onClick', () => {
    // const onPressMock = jest.fn();
    const {getByTestId, getAllByA11yStates} = render(<Login />);
    const element = getByTestId('button-login');
    // act(() => fireEvent.press(element));
    // expect(element.props.onClick).toHaveBeenCalledTimes(1);
    expect(fireEvent.press(element));
  });
});
