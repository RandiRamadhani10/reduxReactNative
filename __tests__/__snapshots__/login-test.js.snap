// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render login screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/logoGedebook.png",
      }
    }
    style={
      Object {
        "height": 525,
        "width": 525,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    LOGIN
  </Text>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "width": 600,
      }
    }
  >
    <TextInput
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "borderWidth": 1,
          "height": 40,
          "margin": 12,
          "padding": 10,
          "width": "100%",
        }
      }
      testID="email-input"
    />
    <TextInput
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "borderWidth": 1,
          "height": 40,
          "margin": 12,
          "padding": 10,
          "width": "100%",
        }
      }
      testID="password-input"
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 30,
          "margin": 5,
          "opacity": 1,
          "padding": 10,
          "width": 225,
        }
      }
      testID="button-login"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "marginTop": 30,
      }
    }
  >
    Dont have account ?
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Register
    </Text>
  </View>
</View>
`;
